src/main/java/com/mraof/minestuck/item/weapon/CustomBow.java

```diff
@@ -0,0 +1,133 @@
+package com.mraof.minestuck.item.weapon;
+
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.enchantment.Enchantments;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.projectile.AbstractArrowEntity;
+import net.minecraft.item.*;
+import net.minecraft.stats.Stats;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.world.World;
+
+public class CustomBow extends BowItem
+{
+	private final Type type;
+	
+	public enum Type
+	{
+		FLAME, LONG, NO_GRAVITY, QUICK, RANDOM_SECOND
+	}
+	
+	public CustomBow(Item.Properties builder, Type type)
```

Comment:
At the moment, this uses the same code as the traditional BowItem class. All it adds is the Type enum where custom material is inserted